src/transformStateWithClones.js

```diff
@@ -7,7 +7,42 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateModified = [];
+  const stateObject = { ...state };
+
+  for (const obj of actions) {
+    for (const key in obj) {
+      if (obj[key] === 'addProperties') {
+        Object.assign(stateObject, obj.extraData);
+
+        const addedObject = { ...stateObject };
+
+        stateModified.push(addedObject);
```

Comment:
```suggestion
        stateModified.push({ ...stateObject });
```
There is no need to create an extra variable.
Also, pay attention to the checklist
![image](https://github.com/mate-academy/js_stateful-clones/assets/96409563/7e365062-8c63-4d4d-a1a7-83cfb5b97c6b)